src/main/java/fishstock/Ui.java

```diff
@@ -0,0 +1,93 @@
+package fishstock;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import fishstock.FishStock.Command;
+
+/**
+ * Encapsulates a Ui.
+ * Handles all input/output to the user.
+ */
+class Ui {
+    protected static void printMsg(String msg) {
+        System.out.println(msg);
+    }
+
+    protected static void printError(String error) {
+        System.out.println(error);
+    }
+
+    /**
+     * The main loop for i/o.
+     * @param list The TaskList for this loop.
+     */
+    protected static void run(TaskList list) {
+        String line = "____________________________________________________________";
+        Scanner sc = new Scanner(System.in);
+        String input;
+        ArrayList<Task> result;
+        boolean hasStopped = false;
+        Task task;
+
+        Ui.printMsg(line + "\nHello, I'm FishStock.\nI might help if I feel like it.");
+
+        while (!hasStopped) {
+            Ui.printMsg(line + "\n");
+            input = sc.nextLine();
+            Ui.printMsg(line);
+
+            Command command = Parser.parse(input);
+
+            try {
+                switch (command) {
+                case BYE:
+                    hasStopped = true;
+                    break;
+                case LIST:
+                    result = list.getList();
```

Comment:
Consider abstracting out this functionality into seperate functions (ie `listTasks / markTask`)!